### PR TITLE
BgInibsMovebg OK

### PR DESCRIPTION
**Pull request by retro-git**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Documentation highly lacking but all matching

### CODE REVIEW

**Comment by EllipticEllipsis** [comment]
Jenkins is complaining because you need to run the format script on this file. You can see this by clicking on the check stuff and looking at the console output.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 1]
```diff
@@ -27,17 +26,51 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
...
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
 };
 
-#endif
+void BgInibsMovebg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgInibsMovebg* this = (BgInibsMovebg*)thisx;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 1);
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:40]:
```suggestion
    BgInibsMovebg* this = THIS;
    
    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
    DynaPolyActor_Init(&this->dyna, 1);
```

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.h [hunk 2]
```diff
@@ -5,9 +5,13 @@
 
 struct BgInibsMovebg;
 
+#define BGINIBSMOVEBG_GET_F(thisx) (u16)(thisx.actor.params) & 0xF
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.h:8]:
```suggestion
#define BGINIBSMOVEBG_GET_F(thisx) ((u16)((thisx)->params) & 0xF)
```

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 3]
```diff
@@ -27,17 +26,51 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
...
+
+    //for some reason doesn't match if I don't use the THIS macro twice here
+    THIS->unk_15C = D_80B96560[BGINIBSMOVEBG_GET_F(this->dyna)];
+    this->unk_160 = D_80B96568[BGINIBSMOVEBG_GET_F(this->dyna)];
+    this->unk_164 = D_80B96570[BGINIBSMOVEBG_GET_F(THIS->dyna)];
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:45]:
```suggestion
    this->unk_15C = D_80B96560[BGINIBSMOVEBG_GET_F(thisx)];
    this->unk_160 = D_80B96568[BGINIBSMOVEBG_GET_F(thisx)];
    this->unk_164 = D_80B96570[BGINIBSMOVEBG_GET_F(thisx)];
```
I find that when THIS needs to be used in some format, usually for regalloc, it indicates the direct actor `thisx` was used.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.h [hunk 4]
```diff
@@ -5,9 +5,13 @@
...
+
 typedef struct BgInibsMovebg {
-    /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x24];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ s32 unk_15C;
+    /* 0x160 */ s32 unk_160;
+    /* 0x164 */ s32 unk_164;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.h:14]:
```suggestion
    /* 0x15C */ Gfx* unk_15C;
    /* 0x160 */ Gfx* unk_160;
    /* 0x164 */ AnimatedMaterial* unk_164;
```
Based on the function calls in `BgInibsMovebg_Draw` we can deduce these types.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 5]
```diff
@@ -27,17 +26,51 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
     (ActorFunc)BgInibsMovebg_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B96578[] = {
+s32 D_80B96560[] = { 0x060062D8, 0x06001DC0 };
+s32 D_80B96568[] = { 0x06006140, 0x06001C10 };
+s32 D_80B96570[] = { 0x06006858, 0x06002598 };
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:31]:
```suggestion
Gfx* D_80B96560[] = { 0x060062D8, 0x06001DC0 };
Gfx* D_80B96568[] = { 0x06006140, 0x06001C10 };
AnimatedMat* D_80B96570[] = { 0x06006858, 0x06002598 };
```
Again the symbols can be deduced from the function calls. Should also convert the raw hex addresses into symbols. They would also needed to be added to undef_syms if they are not already there. For example of this you can take a look at `ObjEnding` which is actually really similar to this.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 6]
```diff
@@ -27,17 +26,51 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
...
+
+    //for some reason doesn't match if I don't use the THIS macro twice here
+    THIS->unk_15C = D_80B96560[BGINIBSMOVEBG_GET_F(this->dyna)];
+    this->unk_160 = D_80B96568[BGINIBSMOVEBG_GET_F(this->dyna)];
+    this->unk_164 = D_80B96570[BGINIBSMOVEBG_GET_F(THIS->dyna)];
+}
+
+void BgInibsMovebg_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, THIS->dyna.bgId);
+}
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:50]:
```suggestion
void BgInibsMovebg_Destroy(Actor* thisx, GlobalContext* globalCtx) {
    BgInibsMovebg* this = THIS;

    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
}
```
Generally always cast with `THIS` then just use `this`, unless of course it does not match.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 7]
```diff
@@ -27,17 +26,51 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Inibs_Movebg/BgInibsMovebg_Destroy.s")
+    new_var4 = new_var5->unk_160;
+    if (((BgInibsMovebg*)thisx)->unk_160 != 0) {
+        func_800BE03C(globalCtx, (void*)new_var4);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Inibs_Movebg/BgInibsMovebg_Draw.s")
+    if (new_var5->unk_160 != 0) {}
+}
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:76]:
```c
void BgInibsMovebg_Draw(Actor* thisx, GlobalContext* globalCtx) {
    BgInibsMovebg* this = THIS;
    AnimatedMaterial* animMat;
    Gfx* dl1;
    Gfx* dl2;

    animMat = this->unk_164;
    if (animMat != 0) {
        AnimatedMat_Draw(globalCtx, Lib_SegmentedToVirtual(this->unk_164));
    }

    dl1 = this->unk_15C;
    if (dl1 != NULL) {
        func_800BDFC0(globalCtx, this->unk_15C);
    }

    dl2 = this->unk_160;
    if (dl2 != NULL) {
        func_800BE03C(globalCtx, this->unk_160);
    }
}
```
This looks to be a cleaner match.

**Review by hensldm — CHANGES_REQUESTED** [review]
Nice work. I was interested so checked out the PR to test a few things and possibly find some cleaner matches, hope you don't mind. I applaud you on `BgInibsMovebg_Draw`, that thing was a doozy.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 8]
```diff
@@ -27,17 +26,47 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
     (ActorFunc)BgInibsMovebg_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B96578[] = {
+Gfx* D_80B96560[] = { 0x060062D8, 0x06001DC0 };
+Gfx* D_80B96568[] = { 0x06006140, 0x06001C10 };
+AnimatedMaterial* D_80B96570[] = { 0x06006858, 0x06002598 };
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:31]: This pointers are segmented pointers. Those refer to assets of this actor's object (you can tell by the `0x06` part of the pointer)
We like to use `extern`ed symbols for those
```suggestion
extern Gfx D_060062D8[];
extern Gfx D_06001DC0[];
extern Gfx D_06006140[];
extern Gfx D_06001C10[];
extern AnimatedMaterial D_06006858;
extern AnimatedMaterial D_06002598;

Gfx* D_80B96560[] = { D_060062D8, D_06001DC0 };
Gfx* D_80B96568[] = { D_06006140, D_06001C10 };
AnimatedMaterial* D_80B96570[] = { &D_06006858, &D_06002598 };
```

You'll need to add those to undefined_syms.txt too

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 9]
```diff
@@ -27,17 +26,56 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
...
+    BgInibsMovebg* this = THIS;
 
-extern InitChainEntry D_80B96578[];
+    AnimatedMaterial* animMat;
+    Gfx* dl1;
+    Gfx* dl2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Inibs_Movebg/BgInibsMovebg_Init.s")
+    animMat = this->unk_164;
+    if (animMat != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:68]:
```suggestion
    if (animMat != NULL) {
```
Since `aninMat` is a pointer.

**Review by hensldm — APPROVED** [review]
Just one small nit. Not enough to stop an approval.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 10]
```diff
@@ -27,17 +26,56 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
...
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B96578[] = {
+extern Gfx D_060062D8[];
+extern Gfx D_06001DC0[];
+extern Gfx D_06006140[];
+extern Gfx D_06001C10[];
+extern AnimatedMaterial D_06006858;
+extern AnimatedMaterial D_06002598;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:34]: If these are all from the same file, can you sort them by address, please? It'll make it easier for the person doing the object.

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c [hunk 11]
```diff
@@ -27,17 +26,56 @@ const ActorInit Bg_Inibs_Movebg_InitVars = {
...
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void BgInibsMovebg_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgInibsMovebg* this = THIS;
 
-extern InitChainEntry D_80B96578[];
+    AnimatedMaterial* animMat;
+    Gfx* dl1;
+    Gfx* dl2;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:65]: Would prefer that these were not so abbreviated, although they'll get named properly once the unks are identified.

File: undefined_syms.txt [hunk 12]
```diff
@@ -858,6 +858,15 @@ D_06013058 = 0x06013058;
...
 D_060016DC = 0x060016DC;
 
+// ovl_Bg_Inibs_Movebg
+
+D_060062D8 = 0x060062D8;
+D_06001DC0 = 0x06001DC0;
+D_06006140 = 0x06006140;
+D_06001C10 = 0x06001C10;
+D_06006858 = 0x06006858;
+D_06002598 = 0x06002598;
```
Comment (EllipticEllipsis) [inline undefined_syms.txt:868]: Sort these too, please.

**Review by EllipticEllipsis — COMMENTED** [review]
Good job on this one, couple of minor things.

**Review by EllipticEllipsis — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c (hunk 11 above)
Comment (engineer124) [reply src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:65]: Agreed, I would prefer `dList` at the very least, so you can what it is at first glance

Comment (Kenix3) [reply src/overlays/actors/ovl_Bg_Inibs_Movebg/z_bg_inibs_movebg.c:65]: Documt will need to name everything, so these will be picked up later. No need to hold up merging for this.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**